client/src/components/runs/logs/forms/ShareWithForm.js

```diff
@@ -201,25 +174,30 @@ export default class ShareWithForm extends React.Component {
   };
 
   onSelectGroup = async () => {
+    const {groupSearchString} = this.state;
     const [role] = this.props.roles
-      .filter(r => !r.predefined && this.splitRoleName(r.name) === this.selectedGroup);
-    const roleName = role ? role.name : this.selectedGroup;
+      .filter(r => !r.predefined && this.splitRoleName(r.name) === groupSearchString.trim());
+    const roleName = role ? role.name : groupSearchString.trim();
```

Comment:
We should check if `groupSearchString` is `null`/`undefined`

Reply:
Done!